### DISCUSSION TITLE
Error handling while sending files through the destination folder

### DISCUSSION THREAD

**Discussion by marc-kuijpers** [drakkan/sftpgo#843] [Q&A]
Hi,

Is SFTPGo responsible for guaranteering the delivery of the file to the destination folder? For a file transfer, can I set the number of retries and the frequency retries are executed? Or is the "guaranteed delivery"-responsibility offloaded through the use of webhooks?

Cheers,
Marc

**Comment by drakkan** [comment]
Hi,

I'm not sure to understand your question. SFTPGo is a server that receives files sent by a client. If there is an error it is the client who must retry the transfer. 

After an event, for example an upload, SFTPGo can notify an external HTTP server. In this case SFTPGo is an HTTP client and if a recoverable error occurs it tries to send the notification again based on the settings [here](https://github.com/drakkan/sftpgo/blob/main/sftpgo.json#L319)
